pkg/controllers/leaderworkerset_controller.go

```diff
@@ -229,7 +248,11 @@ func (r *LeaderWorkerSetReconciler) rollingUpdateParameters(ctx context.Context,
 
 	// Case 2:
 	// Indicates a new rolling update here.
-	if templateUpdated(sts, lws) {
+	hasTemplateUdated, err := templateUpdated(ctx, r.Client, sts, lws)
```

Comment:
Call this in Reconcile and pass to this function a boolean indicating whether or not an update was observed.